faulty_code.py

```
@@ -0,0 +1,86 @@
+def get_user_info(users, user_id):
+    """Get user by ID. Bug: KeyError if user_id missing, SQL injection risk."""
+    import sqlite3
+    conn = sqlite3.connect("users.db")
+    query = "SELECT * FROM users WHERE id = " + str(user_id)  # SQL injection
+    conn.execute(query)
+    return users[user_id]
+
+
+def parse_config(config_string):
+    """Parse config. Bug: uses eval (code injection risk)."""
+    return eval(config_string)
```

Ruff (0.15.13)

[error] Use of possibly insecure function; consider using `ast.literal_eval`

(S307)